src/components/NewWorkspaceModal.js

```diff
@@ -109,86 +99,135 @@ export default ajaxCaller(class NewWorkspaceModal extends Component {
-      title: cloneWorkspace ? 'Clone a Workspace' : 'Create a New Workspace',
-      onDismiss,
-      okButton: buttonPrimary({
-        disabled: errors,
-        tooltip: Utils.summarizeErrors(errors),
-        onClick: () => this.create()
-      }, cloneWorkspace ? 'Clone Workspace' : 'Create Workspace')
-    }, [
-      h(RequiredFormLabel, ['Workspace name']),
-      validatedInput({
-        inputProps: {
-          autoFocus: true,
-          placeholder: 'Enter a name',
-          value: name,
-          onChange: e => this.setState({ name: e.target.value, nameModified: true })
-        },
-        error: Utils.summarizeErrors(nameModified && errors && errors.name)
-      }),
-      h(RequiredFormLabel, ['Billing project']),
-      billingProjects && !billingProjects.length ? h(Fragment, [
-        div({ style: { color: colors.red[0] } }, [
-          icon('error', { size: 16 }),
-          ' You must have a billing project associated with your account to create a new workspace.'
+
+    return Utils.cond(
+      [loading, spinnerOverlay],
+      [hasBillingProjects, () => h(Modal, {
+        title: cloneWorkspace ? 'Clone a Workspace' : 'Create a New Workspace',
+        onDismiss,
+        okButton: buttonPrimary({
+          disabled: errors,
+          tooltip: Utils.summarizeErrors(errors),
+          onClick: () => this.create()
+        }, cloneWorkspace ? 'Clone Workspace' : 'Create Workspace')
+      }, [
+        h(RequiredFormLabel, ['Workspace name']),
+        validatedInput({
+          inputProps: {
+            autoFocus: true,
+            placeholder: 'Enter a name',
+            value: name,
+            onChange: e => this.setState({ name: e.target.value, nameModified: true })
+          },
+          error: Utils.summarizeErrors(nameModified && errors && errors.name)
+        }),
+        h(RequiredFormLabel, ['Billing project']),
+        h(Select, {
+          isClearable: false,
+          placeholder: 'Select a billing project',
+          value: namespace,
+          onChange: ({ value }) => this.setState({ namespace: value }),
+          options: _.uniq(_.map('projectName', billingProjects)).sort()
+        }),
+        h(FormLabel, ['Description']),
+        h(TextArea, {
+          style: { height: 100 },
+          placeholder: 'Enter a description',
+          value: description,
+          onChange: v => this.setState({ description: v })
+        }),
+        h(FormLabel, [
+          'Authorization domain',
+          h(InfoBox, [
+            'An authorization domain can only be set when creating a workspace. ',
+            'Once set, it cannot be changed. ',
+            'Any cloned workspace will automatically inherit the authorization domain(s) from the original workspace and cannot be removed. ',
+            link({
+              href: 'https://broadinstitute.zendesk.com/hc/en-us/articles/360026775691-Managing-Data-Privacy-and-Access-with-Authorization-Domains',
+              target: '_blank'
```

Comment:
reminder to change all the `target: '_blank'`s to the new Utils for safe new tabs!